src/main/java/com/artipie/npm/proxy/NpmProxy.java

```diff
@@ -0,0 +1,361 @@
+
+    /**
+     * The storage.
+     */
+    private final RxStorage storage;
+
+    /**
+     * Web client.
+     */
+    private final WebClient client;
+
+    /**
+     * Ctor.
+     * @param settings Adapter settings
+     * @param vertx Vertx instance
+     * @param storage Adapter storage
+     */
+    public NpmProxy(final NpmProxySettings settings, final Vertx vertx, final Storage storage) {
+        this.settings = settings;
+        this.vertx = vertx;
+        this.storage = new RxStorageWrapper(storage);
+        this.client =  WebClient.create(vertx, NpmProxy.defaultWebClientOptions());
+    }
+
+    /**
+     * Retrieve package metadata.
+     * @param pkg Package name
+     * @return Package metadata (cached or downloaded from remote repository)
+     * @checkstyle ReturnCountCheck (42 lines)
+     */
+    public Maybe<NpmPackage> getPackage(final String pkg) {
+        final Key key = new Key.From(pkg, "package.json");
+        final RequestOptions request = this.buildRequest(pkg);
+        return this.client.request(HttpMethod.GET, request)
+            .timeout(NpmProxy.REQUEST_TIMEOUT)
+            .rxSend()
+            .flatMapMaybe(
+                response -> {
+                    //@checkstyle MagicNumberCheck (1 line)
+                    if (response.statusCode() == 200) {
+                        return Completable.concatArray(
+                            this.storage.save(
+                                key,
+                                new Content.From(
+                                    NpmProxy.internalFormat(response, pkg)
+                                        .getBytes(StandardCharsets.UTF_8)
+                                )
+                            ),
+                            this.storage.save(
+                                new Key.From(pkg, "package.metadata"),
+                                new Content.From(
+                                    NpmProxy.packageMetadata(response)
+                                        .getBytes(StandardCharsets.UTF_8)
+                              )
+                            )
+                        ).andThen(Maybe.defer(() -> this.readPackageFromStorage(pkg).toMaybe()));
+                    } else {
+                        return this.tryLocal(pkg);
+                    }
+                }
+            ).onErrorResumeNext(
+                throwable -> {
+                    Logger.error(
+                        NpmProxy.class,
+                        "Error occurred when process get package call: %s",
+                        throwable.getMessage()
+                    );
+                    return this.tryLocal(pkg);
+                }
+            );
+    }
+
+    /**
+     * Retrieve asset.
+     * @param path Asset path
+     * @return Asset data (cached or downloaded from remote repository)
+     * @checkstyle ReturnCountCheck (54 lines)
+     */
+    public Maybe<NpmAsset> getAsset(final String path) {
```

Comment:
@brastak here "get" prefix also can be omitted

Reply:
It should be consistent with get package method. If the first will be renamed, I rename this

Reply:
@olegmoz ok